### PR TITLE
Added usage guide to main binary and build guide in readme.md

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: README.md
Comment (Wunkolo) [inline README.md]: vkblam does not actually need the renderdoc library to be compiled as it is purely used for my development environment to debug stuff on the GPU. Currently only debug-builds on non-windows platforms technically require it to keep it out of the way of any binary releases as this is the primary development-environment.

Comment (ghost) [reply README.md]: Okay I removed this from the README.md, should be ready for merge now then

**Comment by ghost** [comment]
Should be good to merge now

**Comment by Wunkolo** [comment]
Might hold off on this in favor of a pattern where it will just automatically detect a `bitmaps.map` file in the same directory as the input map file so that the usage is just a list of .map files for easier batch-processing and unit-testing of multiple maps at once.

**Comment by ghost** [comment]
@Wunkolo Can you merge now please? I added a settings json file and improved the README

File: README.md
Comment (Wunkolo) [inline README.md]: This can be done with
```
git clone --recursive https://github.com/Wunkolo/vkblam
cd vkblam
```

File: README.md
Comment (Wunkolo) [inline README.md]: You shouldn't assume the user's build system to be `make`. CMake provides a way to build using whatever the user configured as their default build system
A better process would be:
```
mkdir build
cd build
cmake ..
cmake --build .
```

File: external/.gitignore
Comment (Wunkolo) [inline external/.gitignore]: This isn't related to this PR's intent and shouldn't be here. In general it's better to just ignore the entire `.vscode` directory at the top-level project root. For this PR though there is no need to add this. Especially to the `external/` folder when it can just be added at the top-level.

File: source/main.cpp
Comment (Wunkolo) [inline source/main.cpp]: I do not use `int` but instead I use [standard integer types](https://en.cppreference.com/w/cpp/types/integer) such as `std::uint16_t`.

File: source/main.cpp
Comment (Wunkolo) [inline source/main.cpp]: These variable names do not match the [PascalCase](https://techterms.com/definition/pascalcase) naming pattern found in the rest of the code base.

File: source/main.cpp
Comment (Wunkolo) [inline source/main.cpp]: Be sure to run `.clang-format` on your code to fix style issues like this. I use [Allman-style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style) bracing. `clang-format` will fix all of that for you.

**Review by Wunkolo — CHANGES_REQUESTED** [review]
This PR is containing changes outside its original intent of providing usage-text and build-steps but now contains a json settings submodule integration for the render-size. It also is not following the style-guidance provided in the `.clang-format` file and is not following the pattern of using [standard integer types](https://en.cppreference.com/w/cpp/types/integer) like the rest of the code base or follow the same naming conventions. It would be better to keep PRs atomic in their intent for easy [bisecting](https://git-scm.com/docs/git-bisect) of issues. 

> Can you merge now please?

Please do not insist on merging a PR. You may ask for a re-review but insisting code be merged on someone's personal project can be invalidating of people's boundaries.

**Comment by Wunkolo** [comment]
I think you may have incidentally removed all of the submodules except the json one. The builds fail without them.
image

File: README.md
Comment (ghost) [reply README.md]: image
I couldn't clone the sub-modules due to presumably a configuration issue. I honestly have never needed to work with them before today so I'm unsure how to correct that problem.

File: README.md
Comment (ghost) [reply README.md]: Good point, personally just never thought of that. Ill push that as a patch to my fork then resolve this conversation.

Comment (ghost) [reply README.md]: Done as of commit [ce2b26b](https://github.com/Wunkolo/vkblam/pull/1/commits/ce2b26beb019a45767eaa2389a6fee01cc390b2d)

File: source/main.cpp
Comment (ghost) [reply source/main.cpp]: Fixed in [bbc8649](https://github.com/Wunkolo/vkblam/pull/1/commits/bbc8649cf4c93550cbd1a16611d7e3a3c108c38b). Closing.

**Comment by ghost** [comment]
> I think you may have incidentally removed all of the submodules except the json one. The builds fail without them. image

Fixed in [bbc8649](https://github.com/Wunkolo/vkblam/pull/1/commits/bbc8649cf4c93550cbd1a16611d7e3a3c108c38b).

File: source/main.cpp
Comment (ghost) [reply source/main.cpp]: Fixed in [bbc8649](https://github.com/Wunkolo/vkblam/pull/1/commits/bbc8649cf4c93550cbd1a16611d7e3a3c108c38b). Closing.

File: source/main.cpp
Comment (ghost) [reply source/main.cpp]: Fixed in [bbc8649](https://github.com/Wunkolo/vkblam/pull/1/commits/bbc8649cf4c93550cbd1a16611d7e3a3c108c38b). Closing.

File: external/.gitignore
Comment (ghost) [reply external/.gitignore]: Fixed in [bbc8649](https://github.com/Wunkolo/vkblam/pull/1/commits/bbc8649cf4c93550cbd1a16611d7e3a3c108c38b). Closing.

File: README.md
Comment (ghost) [reply README.md]: Fixed in [bbc8649](https://github.com/Wunkolo/vkblam/pull/1/commits/bbc8649cf4c93550cbd1a16611d7e3a3c108c38b) and [f8b1f0e](https://github.com/Wunkolo/vkblam/pull/1/commits/f8b1f0e4b15c1cdc7e90eca557d20fa522826ec1). Closing.

**Comment by Wunkolo** [comment]
I think the scope of changes within this PR and required changes before this becomes merge-ready for the original intent are starting to become too great. To save everyone time I think I'll just close this PR and will likely just implement the usage-message and build-documentation myself.

**Comment by ghost** [comment]
What are you on about man? It all works well, I've formatted it and made the changes you asked for. Infact I even fixed some existing bugs like the recursive clone not working. 👍

I get that the changes in this PR have become a lot more than in the title but I'm confused and a bit hurt about why it's been rejected when it does exactly what it needs to, with the formatting you requested. @Wunkolo

**Comment by Wunkolo** [comment]
It does not build:
https://github.com/Wunkolo/vkblam/runs/8073829417?check_suite_focus=true#step:5:40
https://github.com/Wunkolo/vkblam/runs/8073829574?check_suite_focus=true#step:5:154

I mentioned the issues in my [latest review](https://github.com/Wunkolo/vkblam/pull/1/files/f8b1f0e4b15c1cdc7e90eca557d20fa522826ec1). `jsonpp` is not added as a submodule or added properly into the CMakeLists.txt. It's also not related to the intent of this PR as I am trying to keep these things atomic. I don't want a usage-guide and build-guide PR to have an entirely new json configuration system embedded into it. You have changed all of the submodules from `ssh` to `https`, and added an extra redundant `.gitmodules` file to the externals folder. I think it's better to have separate PRs for the two additions. As it is now, it's breaking.

**Comment by ghost** [comment]
Good point actually, I assumed that the user would have the json library installed on their system. Although that would definitely not be the case for windows.

**Comment by ghost** [comment]
As for the submodles I just re-added them back and they work now, whereas with the previous ssh type it wasn't. Anyways I don't personally see any need for ssh to download public library's instead of https so i dont think that particular needs changing. 👍

**Pull request closed without merging**
